common/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/view/listitem/OneLineListItem.kt

```diff
@@ -50,152 +63,44 @@ class OneLineListItem @JvmOverloads constructor(
             R.style.Widget_DuckDuckGo_OneLineListItem,
         ).apply {
 
-            binding.primaryText.text = getString(R.styleable.OneLineListItem_primaryText)
+            setPrimaryText(getString(R.styleable.OneLineListItem_primaryText))
 
             if (hasValue(R.styleable.OneLineListItem_primaryTextColorOverlay)) {
-                binding.primaryText.setTextColor(getColorStateList(R.styleable.OneLineListItem_primaryTextColorOverlay))
+                setPrimaryTextColorStateList(getColorStateList(R.styleable.OneLineListItem_primaryTextColorOverlay))
             }
 
             val truncated = getBoolean(R.styleable.OneLineListItem_primaryTextTruncated, false)
-            if (truncated) {
-                binding.primaryText.maxLines = 1
-                binding.primaryText.ellipsize = TruncateAt.END
-            }
+            setPrimaryTextTruncation(truncated)
 
             if (hasValue(R.styleable.OneLineListItem_leadingIcon)) {
                 setLeadingIconDrawable(getDrawable(R.styleable.OneLineListItem_leadingIcon)!!)
             } else {
-                binding.leadingIconBackground.gone()
+                setLeadingIconVisibility(false)
             }
 
             if (hasValue(R.styleable.OneLineListItem_leadingIconBackground)) {
                 setLeadingIconBackgroundType(getInt(R.styleable.OneLineListItem_leadingIconBackground, 0))
             }
 
+            if (hasValue(R.styleable.OneLineListItem_leadingIconSize)) {
+                val imageSize = LeadingIconSize.from(getInt(R.styleable.OneLineListItem_leadingIconSize, 1))
+                setLeadingIconSize(imageSize)
+            }
+
             val showTrailingIcon = hasValue(R.styleable.OneLineListItem_trailingIcon)
             val showSwitch = getBoolean(R.styleable.OneLineListItem_showSwitch, false)
             when {
                 showSwitch -> showSwitch()
                 showTrailingIcon -> {
-                    binding.trailingIcon.setImageDrawable(getDrawable(R.styleable.OneLineListItem_trailingIcon))
+                    setTrailingIconDrawable(getDrawable(R.styleable.OneLineListItem_trailingIcon)!!)
                     showTrailingIcon()
                 }
                 else -> {
-                    binding.trailingIconContainer.gone()
-                    binding.trailingSwitch.gone()
+                    hideTrailingItems()
```

Comment:
NIT: Use `hideTrailingItems()` just like you did here for _TwoLineListItem_ as well (_Line 112_)